speech/google/cloud/speech/streaming/container.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2016 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Representation of a group of GAPIC Speech API responses."""
+
+from google.cloud.speech.streaming.response import StreamingSpeechResponse
+
+
+class StreamingResponseContainer(object):
+    """Response container to help manage streaming responses.
+
+    :type responses: list of :class:`~response.StreamingSpeechResponse`
+    :param responses: List of ``StreamingSpeechResponse`` objects.
+    """
+    def __init__(self, responses=None):
+        self._responses = responses or {}
+
+    def add_response(self, response):
+        """Add/update response based on the ``result_index``.
+
+        :type response: :class:`~response.StreamingSpeechResponse`
+        :param response: Instance of ``StreamingSpeechResponse``.
+        """
+        self._responses.update({response.result_index:
+                                StreamingSpeechResponse.from_pb(response)})
+
+    @property
+    def responses(self):
+        """All responses held in container.
+
+        :rtype: list of :class:`~response.StreamingSpeechResponse`
+        :returns: List of ``StreamingSpeechResponse`` objects.
+        """
+        return self._responses
```

Comment:
Using a mutable object as a read-only object is a recipe for failure. You should return a **copy** here.